boxtree/tree.py

```diff
@@ -111,6 +111,83 @@ class box_flags_enum(Enum):  # noqa
 # }}}
 
 
+# {{{ tree of boxes
+
+class TreeOfBoxes:
```

Comment:
Would it be possible to make this a base class of `Tree`?

Reply:
As discussed in https://github.com/inducer/boxtree/pull/35#discussion_r625822132, `TreeOfBoxes` is not suitable as the base class. A viable alternative may be letting both `Tree` and `TreeOfBoxes` share one base class.

Reply:
b7c0d30983678e345757c16060ddca8033ef6212